tests/fixed_point/test_fp_arithmetic_sugar.py

```diff
@@ -188,6 +126,34 @@ def test_sub_fail(self):
         with self.assertRaises(TypeError):
             _ = int_value - fixed_point_value  # type: ignore
 
+    def test_sub_nonfinite(self):
```

Comment:
I know this isn't part of the review, but on line 125 where you test for type errors, I would argue it's reasonable to internally convert integers to fixed points since there's no loss of precision (very similar to how native python supports float \op int). I see a point with not supporting float + fixed point due.